### PR TITLE
avm2: Support indexing with QNames

### PR DESCRIPTION
**Pull request by adrian17**
(draft, as I have yet to write some tests.)

### CODE REVIEW

**Comment by adrian17** [comment]
Un-WIP'd.

File: core/src/avm2/multiname.rs [hunk 1]
```diff
@@ -220,12 +221,37 @@ impl<'gc> Multiname<'gc> {
...
+    #[inline(never)]
+    #[cold]
+    pub fn try_replace_with_qname(
+        &self,
+        obj: Object<'gc>,
+        activation: &mut Activation<'_, 'gc, '_>,
+    ) -> Option<Self> {
+        if let Object::QNameObject(qname_object) = obj {
+            if self.has_lazy_ns() {
+                activation.pop_stack(); // ignore the ns component on stack
```
Comment (Herschel) [inline core/src/avm2/multiname.rs:233]: style nit, `let _ = activation.pop_stack();` is a little more explicit about ignoring this

Comment (adrian17) [reply core/src/avm2/multiname.rs:233]: In that case, should we add `#[must_use]` to the function?

Comment (Herschel) [reply core/src/avm2/multiname.rs:233]: Yep! I agree

**Review by Herschel — APPROVED** [review]
Thank you!

**Pull request merged**